host/3.0/publish.yml

```diff
@@ -1,555 +1,606 @@
-      docker push $TARGET_REGISTRY/python:$(TargetVersion)-python3.9-buildenv
-
-
-      docker push $TARGET_REGISTRY/python:3.0
-      docker push $TARGET_REGISTRY/python:3.0-slim
-      docker push $TARGET_REGISTRY/python:3.0-appservice
-      docker push $TARGET_REGISTRY/python:3.0-appservice-quickstart
-      docker push $TARGET_REGISTRY/python:3.0-buildenv
-
-      docker push $TARGET_REGISTRY/python:3.0-python3.6
-      docker push $TARGET_REGISTRY/python:3.0-python3.6-slim
-      docker push $TARGET_REGISTRY/python:3.0-python3.6-appservice
-      docker push $TARGET_REGISTRY/python:3.0-python3.6-appservice-quickstart
-      docker push $TARGET_REGISTRY/python:3.0-python3.6-buildenv
-
-      docker push $TARGET_REGISTRY/python:3.0-python3.7
-      docker push $TARGET_REGISTRY/python:3.0-python3.7-slim
-      docker push $TARGET_REGISTRY/python:3.0-python3.7-appservice
-      docker push $TARGET_REGISTRY/python:3.0-python3.7-appservice-quickstart
-      docker push $TARGET_REGISTRY/python:3.0-python3.7-buildenv
-
-      docker push $TARGET_REGISTRY/python:3.0-python3.8
-      docker push $TARGET_REGISTRY/python:3.0-python3.8-slim
-      docker push $TARGET_REGISTRY/python:3.0-python3.8-appservice
-      docker push $TARGET_REGISTRY/python:3.0-python3.8-appservice-quickstart
-      docker push $TARGET_REGISTRY/python:3.0-python3.8-buildenv
-
-      docker push $TARGET_REGISTRY/python:3.0-python3.9
-      docker push $TARGET_REGISTRY/python:3.0-python3.9-slim
-      docker push $TARGET_REGISTRY/python:3.0-python3.9-appservice
-      docker push $TARGET_REGISTRY/python:3.0-python3.9-appservice-quickstart
-      docker push $TARGET_REGISTRY/python:3.0-python3.9-buildenv
-
-      docker system prune -a -f
-    displayName: tag and push python images
-    continueOnError: false
+stages:
+- stage: Linux_Images
+  jobs:
+  - job: Publish_all
+    pool: 
+      vmImage: ubuntu-18.04
+    steps:
+    - bash: |
+        echo $pswd | docker login -u $(dockerUsername) --password-stdin azurefunctions.azurecr.io
+      displayName: login
+      continueOnError: false
+      env:
+        pswd: $(dockerPassword)
+
+    - bash: |
+        set -e
+        SOURCE_REGISTRY=azurefunctions.azurecr.io/azure-functions/3.0
+        TARGET_REGISTRY=azurefunctions.azurecr.io/public/azure-functions
+
+        if [ -z "$(TargetVersion)" ]; then
+          echo "ERROR: TargetVersion is required"
+          exit 1
+        fi
+
+        if [ -z "$(PrivateVersion)" ]; then
+          echo "ERROR: PrivateVersion is required"
+          exit 1
+        fi
+
+        echo "##vso[task.setvariable variable=SOURCE_REGISTRY]$SOURCE_REGISTRY"
+        echo "##vso[task.setvariable variable=TARGET_REGISTRY]$TARGET_REGISTRY"
+      displayName: set env
+      continueOnError: false
+
+    - bash: |
+        set -e
+        docker pull $SOURCE_REGISTRY/dotnet:$(PrivateVersion)
+        docker pull $SOURCE_REGISTRY/dotnet:$(PrivateVersion)-slim
```

Comment:
diff for this file looks quite large. Can you please check if this is related to formatting? if yes, do make sure only changes needed to publish new images show up in the diff

Reply:
I actually did change the format of our pipelines.  I updated them to the most recent Azure Devops Syntax.  Some directives, such as Queue, were outdated and their wiki pages mentioned that they were no longer supported. 
I also introduced syntax to stage the pipeline so that nanoserver images can run in parallel and not slow down our Linux images. This was a suggestion from my conversations with Tsuyoshi.

Reply:
Great thanks for the explanation! Thanks for to @TsuyoshiUshio for the pointer. Resolving this comment.